cctools/ld64/src/ld/Resolver.cpp

```diff
@@ -1192,9 +1192,10 @@ void Resolver::deadStripOptimize(bool force)
 			// unset liveness, so markLive() will recurse
 			(const_cast<ld::Atom*>(atom))->setLive(0);
 		}
-		// <rdar://problem/49468634> if doing LTO, mark all libclang_rt* mach-o atoms as live since the backend may suddenly codegen uses of them
+		// <rdar://problem/49468634> if doing LTO, mark all libclang_rt* mach-o atoms as live since the backend may suddenly codegen uses of them.
+		// Likewise with rust compiler_builtins atoms. They may come from a libcompiler_builtins-<hash>.a lib, or from a compiler_builtins-... member of a rust static library.
 		else if ( _haveLLVMObjs && !force && (atom->contentType() !=  ld::Atom::typeLTOtemporary) ) {
-			if ( isCompilerSupportLib(atom->safeFilePath()) ) {
+			if ( isCompilerSupportLib(atom->safeFilePath()) || strstr(atom->safeFilePath(), "compiler_builtins") != nullptr ) ) {
```

Comment:
Note that I didn't add this check in isCompilerSupportLib because I wasn't sure of the side effects from other code using that function.